### PR TITLE
General Update for 2022 and dynamic dev version support

### LINKED ISSUE #1: The directory needs to be a git repo
I don't think we actually mention you need to have a git repo for setuptools_scm to work.

### LINKED ISSUE #19: Add a `.gitignore`?
We don't automake a git directory, but it is expected (#1), having a ignore would be very useful?

### PR DESCRIPTION
**Pull request by Cadair**
This is a continuation of #13 with some further cleanup.

Fixes #1 by docs
Fixes #19

----

Questions to answer before merge:

1. Do we want setuptools_scm to be optional (it isn't at the moment).  - Guide yes, template no.
2. If yes what do we want the default to be  - N/A
3. Do we want _dev to be optional? - Yes

### CODE REVIEW

**Comment by nabobalis** [comment]
Some beautiful changes, you should thank the person who started the ball on this.

**Comment by bsipocz** [comment]
> Do we want setuptools_scm to be optional (it isn't at the moment). 

not having it would mean that we can have a reasonable dev version number without the awkward tagging after a release, right? If yes, then also I would go with "no" for q2

**Comment by Cadair** [comment]
> without the awkward tagging after a release

Well that's a "feature" of tagging on release branches. I think that minor annoyance is worth the general helpfulness of not having to manually maintain a version number.

I would point out at this point that this is supposed to be an *opinionated* template & guide, everything in the template should be documented and we want the documentation to be easy to follow.

The only middle ground we could do is have a hidden option which defaults to on, so downstream templates could optionally turn it off, although I would rather not do that either. If it's worth having then it's worth having on by default and documenting.

**Comment by Cadair** [comment]
> where releases are done from the default branch

You don't have to do it in this case, you only have to dev tag if you put your release tags on the not default branch.

File: tox.ini [hunk 1]
```diff
@@ -1,27 +1,26 @@
...
 deps =
-	cookiecutter
-	tox
-	pytest-cookies
-	xdist: pytest-xdist
+    cookiecutter
+    pytest-cookies
+    pytest-virtualenv
+    tox[testing]
+    virtualenv==20.19.0
```
Comment (nabobalis) [inline tox.ini:14]: Is there a reason to pin this? And if so, why not the latest version?

File: cookiecutter.json [hunk 2]
```diff
@@ -4,12 +4,23 @@
...
+        "BSD 3-Clause",
+        "GNU GPL v3+",
+        "Apache Software License 2.0",
+        "BSD 2-Clause",
+        "Other"
+    ],
     "project_url": "",
-    "minimum_python_version": ["3.8", "3.7", "3.6", "3.5"],
+    "minimum_python_version": [
+        "3.8",
```
Comment (nabobalis) [inline cookiecutter.json:16]: Should this not be 3.9 these days due to NEP29?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -16,16 +14,10 @@ jobs:
   test:
     uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@v1
     with:
+      toxdeps: "virtualenv==20.19.0"
```
Comment (nabobalis) [inline .github/workflows/tests.yml:17]: Does this need to be pinned here as well?

File: cookiecutter.json [hunk 4]
```diff
@@ -4,12 +4,23 @@
...
+        "3.9",
+        "3.10",
+        "3.11"
```
Comment (Cadair) [inline cookiecutter.json:19]:
```suggestion
        "3.11",
        "3.12",
```

File: .github/workflows/tests.yml (hunk 3 above)
Comment (CyclingNinja) [reply .github/workflows/tests.yml:17]: It did not, sorted

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -14,7 +14,7 @@ jobs:
   test:
     uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@v1
     with:
-      toxdeps: "virtualenv==20.19.0"
+      toxdeps: "virtualenv"
```
Comment (Cadair) [inline .github/workflows/tests.yml:17]: we don't need this line at all.

File: tox.ini [hunk 6]
```diff
@@ -11,7 +11,7 @@ deps =
     pytest-cookies
     pytest-virtualenv
     tox[testing]
-    virtualenv==20.19.0
+    virtualenv
```
Comment (Cadair) [inline tox.ini:14]: or this one for that matter.

**Pull request merged**